CMakeLists.txt

```diff
@@ -69,12 +70,32 @@ else()
     message(STATUS "Configuring Hyprland in Release with CMake")
 endif()
 
-include_directories(
-  .
-  "subprojects/wlroots/include/"
-  "subprojects/wlroots/build/include/"
-  "subprojects/udis86/"
-  "protocols/")
+set(WLROOTS_SOVERSION 12032)
+
+find_program(MESON_EXE NAMES meson)
+find_program(NINJA_EXE NAMES ninja)
+find_program(SED_EXE NAMES sed)
+ExternalProject_Add(wlroots_build
+    PREFIX wlroots
+    SOURCE_DIR ${CMAKE_SOURCE_DIR}/subprojects/wlroots
+    PATCH_COMMAND ${SED_EXE} -E -i -e "s/(soversion = 12)([^032]|$$)/soversion = ${WLROOTS_SOVERSION}/g" ${CMAKE_SOURCE_DIR}/subprojects/wlroots/meson.build
+    CONFIGURE_COMMAND ${MESON_EXE} setup --prefix "/" . ${CMAKE_SOURCE_DIR}/subprojects/wlroots
+    BUILD_COMMAND ${NINJA_EXE} -C .
+    INSTALL_COMMAND ${MESON_EXE} install --destdir ${CMAKE_CURRENT_BINARY_DIR}/wlroots
+	BUILD_BYPRODUCTS ${CMAKE_CURRENT_BINARY_DIR}/wlroots/lib/libwlroots.so.${WLROOTS_SOVERSION})
+
+set(wlroots_INCLUDE_DIR ${CMAKE_SOURCE_DIR}/subprojects/wlroots/include/)
```

Comment:
I guess it should be something like:

```cmake
ExternalProject_Get_property(wlroots_build INSTALL_DIR)
set(wlroots_INCLUDE_DIR ${INSTALL_DIR}/include/)
```
I think that is reason why the build currently fails. `config.h` is a generated file and is either in the build directory or in the final "installation" directory.